include/deal.II/base/polynomials_pyramid.h

```diff
@@ -0,0 +1,166 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+#ifndef dealii_polynomials_pyramid_h
+#define dealii_polynomials_pyramid_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/scalar_polynomials_base.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * A namespace for functions and classes that provide support for simplex
+ * reference cell entities, i.e., triangles and tetrahedrons.
+ *
+ *  @ingroup simplex
+ */
+namespace Simplex
+{
+  /**
+   * Polynomials defined on pyramid entities. This class is basis of
+   * Simplex::FE_PyramidP.
+   */
+  template <int dim>
+  class ScalarPyramidPolynomial : public ScalarPolynomialsBase<dim>
```

Comment:
This may be a separate question from this PR, but the name of the class suggests that it is a *generic* base class for polynomials on pyramids, whereas the constructor takes a `degree` argument that suggests that it is actually meant to implement generic Lagrange elements.

Reply:
Yes. Let's address this in a follow-up PR. What about `ScalarLagrangePyramidPolynomial`?

Reply:
I think it would be useful to keep the words "LagrangePolynomial" together. Maybe `ScalarLagrangePolynomialPyramid` or `...OnPyramids`?